CI fails with dev dependencies: ValueError: The following `model_kwargs` are not used by the model: ['mm_token_type_ids']

CI fails with dev dependencies: https://github.com/huggingface/trl/actions/runs/22501157108/job/65188306952
> ValueError: The following `model_kwargs` are not used by the model: ['mm_token_type_ids'] (note: typos in the generate arguments will also show up in this list)

```python
   FAILED tests/test_grpo_trainer.py::TestGRPOTrainer::test_training_vlm_and_liger[trl-internal-testing/tiny-Qwen2_5_VLForConditionalGeneration] - ValueError: The following `model_kwargs` are not used by the model: ['mm_token_type_ids'] (note: typos in the generate arguments will also show up in this list)
```

Stacktrace:
```python
   >       trainer.train()
  
  tests/test_grpo_trainer.py:2013: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .venv/lib/python3.12/site-packages/transformers/trainer.py:1424: in train
      return inner_training_loop(
  .venv/lib/python3.12/site-packages/transformers/trainer.py:1506: in _inner_training_loop
      self._run_epoch(
  .venv/lib/python3.12/site-packages/transformers/trainer.py:1734: in _run_epoch
      tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:1096: in training_step
      output = super().training_step(model, inputs, num_items_in_batch)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/trainer.py:1900: in training_step
      inputs = self._prepare_inputs(inputs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/extras/profiling.py:202: in wrapper
      return func(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:1125: in _prepare_inputs
      generation_batch = self._generate_and_score_completions(generation_batch)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:1645: in _generate_and_score_completions
      ) = self._generate(prompts)
          ^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:1511: in _generate
      prompt_ids, completion_ids, logprobs, extra_fields = self._generate_single_turn(prompts)
                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:1313: in _generate_single_turn
      prompt_completion_ids = unwrapped_model.generate(
  .venv/lib/python3.12/site-packages/torch/utils/_contextlib.py:124: in decorate_context
      return func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/generation/utils.py:2352: in generate
      self._validate_model_kwargs(model_kwargs.copy())
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = Qwen2_5_VLForConditionalGeneration(
    (model): Qwen2_5_VLModel(
      (visual): Qwen2_5_VisionTransformerPretrainedModel...tary_emb): Qwen2_5_VLRotaryEmbedding()
      )
    )
    (lm_head): Linear(in_features=16, out_features=151936, bias=False)
  )
  model_kwargs = {'attention_mask': tensor([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
           1, 1, 1, ..., 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0,
           0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], device='cuda:0'), ...}
  
      def _validate_model_kwargs(self, model_kwargs: dict[str, Any]):
          """Validates model kwargs for generation. Generate argument typos will also be caught here."""
          # Excludes arguments that are handled before calling any model function
          if self.config.is_encoder_decoder:
              for key in ["decoder_input_ids"]:
                  model_kwargs.pop(key, None)
      
          unused_model_args = []
          model_args = set(inspect.signature(self.prepare_inputs_for_generation).parameters)
          # `kwargs`/`model_kwargs` is often used to handle optional forward pass inputs like `attention_mask`. If
          # `prepare_inputs_for_generation` doesn't accept them, then a stricter check can be made ;)
          if "kwargs" in model_args or "model_kwargs" in model_args:
              model_args |= set(inspect.signature(self.forward).parameters)
      
          # Encoder-Decoder models may also need Encoder arguments from `model_kwargs`
          if self.config.is_encoder_decoder:
              base_model = getattr(self, self.base_model_prefix, None)
      
              # allow encoder kwargs
              encoder = getattr(self, "encoder", None)
              # `MusicgenForConditionalGeneration` has `text_encoder` and `audio_encoder`.
              # Also, it has `base_model_prefix = "encoder_decoder"` but there is no `self.encoder_decoder`
              # TODO: A better way to handle this.
              if encoder is None and base_model is not None:
                  encoder = getattr(base_model, "encoder", None)
      
              if encoder is not None:
                  encoder_model_args = set(inspect.signature(encoder.forward).parameters)
                  model_args |= encoder_model_args
      
              # allow decoder kwargs
              decoder = getattr(self, "decoder", None)
              if decoder is None and base_model is not None:
                  decoder = getattr(base_model, "decoder", None)
      
              if decoder is not None:
                  decoder_model_args = set(inspect.signature(decoder.forward).parameters)
                  model_args |= {f"decoder_{x}" for x in decoder_model_args}
      
          # TransformersKwargs are model-agnostic attention and generation arguments such as 'output_attentions'
          for key, value in model_kwargs.items():
              if (
                  value is not None
                  and key not in model_args
                  and key not in TransformersKwargs.__optional_keys__
                  and key != "debug_io"
              ):
                  unused_model_args.append(key)
      
          if unused_model_args:
  >           raise ValueError(
                  f"The following `model_kwargs` are not used by the model: {unused_model_args} (note: typos in the"
                  " generate arguments will also show up in this list)"
              )
  E           ValueError: The following `model_kwargs` are not used by the model: ['mm_token_type_ids'] (note: typos in the generate arguments will also show up in this list)
  
  .venv/lib/python3.12/site-packages/transformers/generation/utils.py:1541: ValueError
```

--- a/tests/test_grpo_trainer.py
+++ b/tests/test_grpo_trainer.py
@@ -1985,8 +1985,8 @@ class TestGRPOTrainer(TrlTestCase):
                 "trl-internal-testing/tiny-Qwen2_5_VLForConditionalGeneration",
                 marks=pytest.mark.xfail(
                     (Version("5.2.0") < Version(transformers.__version__))
-                    and is_liger_kernel_available(min_version="0.8.0"),
-                    reason=("Upstream issue tracked at https://github.com/linkedin/Liger-Kernel/issues/1117"),
+                    and not is_liger_kernel_available(min_version="0.8.0"),
+                    reason="Upstream issue tracked at https://github.com/linkedin/Liger-Kernel/issues/1117",
                 ),
             ),
         ],

